tests/test_result.py

```diff
@@ -64,6 +63,29 @@ def test_err() -> None:
     assert res.value == ':('
 
 
+def test_is_ok_is_err() -> None:
+    """The module-level check functions work."""
+    o: Result[str, int] = Ok('yay')
+    n: Result[str, int] = Err(100)
```

Comment:
I'm using pyright and it's reporting these as `Ok` instead of `Result`. A simple fix is to define a identify function that makes the typechecker not report these as the literal types,
```
    def passthrough(o: Result[str, int]) -> Result[str, int]:
        return o

    o = passthrough(Ok('yay'))
    n = passthrough(Err(100))
```

Without this passthrough function, you don't get proper type checking in this case,
```def test_is_ok_is_err_type_guard() -> None:
    """The module-level type guard shorthands work."""
    def passthrough(o: Result[str, int]) -> Result[str, int]:
        return o

    o = passthrough(Ok('yay'))
    n = passthrough(Err(100))

    # Type checkers supporting typing.TypeGuard should accept the code below:
    if not is_ok(o):
        _ = o.value + ""  # works only for str
              ^^^^^^^^^^^
              Operator "+" not supported for types "str | int" and "Literal['']" 
    if not is_err(n):
        _ = n.value + 0  # works only for int
              ^^^^^^^^^^^
              Operator "+" not supported for types "str | int" and "Literal[0]"
```